src/CoreShop/Bundle/IndexBundle/Worker/MysqlWorker/Listing.php

```diff
@@ -471,7 +472,7 @@ protected function addQueryFromConditions(QueryBuilder $queryBuilder, $excludeCo
             }
         } else {
             if ($variantMode == AbstractListing::VARIANT_MODE_HIDE) {
-                $queryBuilder->andWhere('o_type != \'variant\'');
+                $queryBuilder->andWhere('q.o_type != \'variant\'');
             }
             if (!$excludeConditions) {
                 $this->addUserSpecificConditions($queryBuilder, $excludedFieldName);
```

Comment:
addUserSpecificConditions():

`$queryBuilder->andWhere('q.o_id IN (SELECT DISTINCT src FROM ' . $relationalTableName . ' q WHERE ' . $cond . ')');`